storage: `TestReplicateRogueRemovedNode` is flaky

The following test appears to have failed:

[#8544](https://circleci.com/gh/cockroachdb/cockroach/8544):

```
E1021 19:20:38.010347 954 storage/client_test.go:497  error reading "a" from engine 1: value type is not INT: UNKNOWN
E1021 19:20:39.011054 954 storage/client_test.go:497  error reading "a" from engine 1: value type is not INT: UNKNOWN
I1021 19:20:40.014157 954 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
I1021 19:20:40.014367 954 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
I1021 19:20:40.014498 954 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- FAIL: TestReplicateRogueRemovedNode (3.39s)
	<autogenerated>:32: storage/client_raft_test.go:1361: condition failed to evaluate within 3s: storage/client_test.go:511: expected [16 0 0], got [16 0 5]
=== RUN   TestReplicateReAddAfterDown
I1021 19:20:40.020060 954 multiraft/multiraft.go:524  node 1 starting
I1021 19:20:40.020440 954 storage/replica.go:1244  gossiping cluster id  from store 1, range 1
I1021 19:20:40.022304 954 raft/raft.go:409  [group 1] 1 became follower at term 5
I1021 19:20:40.022546 954 raft/raft.go:218  [group 1] newRaft 1 [peers: [1], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I1021 19:20:40.022794 954 multiraft/multiraft.go:918  node 1 campaigning because initial confstate is [1]
I1021 19:20:40.023125 954 raft/raft.go:488  [group 1] 1 is starting a new election at term 5
I1021 19:20:40.023253 954 raft/raft.go:422  [group 1] 1 became candidate at term 6
I1021 19:20:40.023341 954 raft/raft.go:471  [group 1] 1 received vote from 1 at term 6
--
1      storage/replica.go:1514
I1021 19:20:44.143610 954 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestStoreRangeSystemSplits (0.72s)
=== RUN   Example_rebalancing
--- PASS: Example_rebalancing (0.87s)
FAIL
FAIL	github.com/cockroachdb/cockroach/storage	24.062s
=== RUN   TestBatchBasics
I1021 19:20:16.256442 934 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.01s)
=== RUN   TestBatchGet
I1021 19:20:16.260942 934 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.01s)
=== RUN   TestBatchMerge
I1021 19:20:16.272093 934 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchMerge (0.01s)
=== RUN   TestBatchProto

```
Please assign, take a look and update the issue accordingly.

--- a/storage/client_raft_test.go
+++ b/storage/client_raft_test.go
@@ -1285,7 +1285,6 @@ func TestStoreRangeRebalance(t *testing.T) {
 // cannot cause other removed nodes to recreate their ranges.
 func TestReplicateRogueRemovedNode(t *testing.T) {
 	defer leaktest.AfterTest(t)
-	t.Skip("TODO(bdarnell): https://github.com/cockroachdb/cockroach/issues/2880")
 
 	mtc := startMultiTestContext(t, 3)
 	defer mtc.Stop()

--- a/storage/store.go
+++ b/storage/store.go
@@ -1399,9 +1399,14 @@ func (s *Store) ProposeRaftCommand(idKey cmdIDKey, cmd roachpb.RaftCommand) <-ch
 
 // proposeRaftCommandImpl runs on the processRaft goroutine.
 func (s *Store) proposeRaftCommandImpl(idKey cmdIDKey, cmd roachpb.RaftCommand) <-chan error {
-	// Lazily create group. TODO(bdarnell): make this non-lazy
-	err := s.multiraft.CreateGroup(cmd.RangeID)
-	if err != nil {
+	// If the range has been removed since the proposal started, drop it now.
+	if _, ok := s.replicas[cmd.RangeID]; !ok {
+		ch := make(chan error, 1)
+		ch <- roachpb.NewRangeNotFoundError(cmd.RangeID)
+		return ch
+	}
+	// Lazily create group.
+	if err := s.multiraft.CreateGroup(cmd.RangeID); err != nil {
 		ch := make(chan error, 1)
 		ch <- err
 		return ch

